src/util/HFactorDebug.cpp

```diff
@@ -96,20 +96,16 @@ void debugReportRankDeficiency(
 
 void debugReportRankDeficientASM(
     const HighsInt highs_debug_level, const HighsLogOptions& log_options,
-    const HighsInt num_row, const vector<HighsInt>& mc_start,
-    const vector<HighsInt>& mc_count_a, const vector<HighsInt>& mc_index,
-    const vector<double>& mc_value, const vector<HighsInt>& iwork,
-    const HighsInt rank_deficiency, const vector<HighsInt>& col_with_no_pivot,
-    const vector<HighsInt>& row_with_no_pivot) {
+    const HighsInt num_row, const std::vector<HighsInt>& mc_start,
+    const std::vector<HighsInt>& mc_count_a,
+    const std::vector<HighsInt>& mc_index, const std::vector<double>& mc_value,
+    const std::vector<HighsInt>& iwork, const HighsInt rank_deficiency,
+    const std::vector<HighsInt>& col_with_no_pivot,
+    const std::vector<HighsInt>& row_with_no_pivot) {
   if (highs_debug_level == kHighsDebugLevelNone) return;
   if (rank_deficiency > 10) return;
-  double* ASM;
-  ASM = (double*)malloc(sizeof(double) * rank_deficiency * rank_deficiency);
-  for (HighsInt i = 0; i < rank_deficiency; i++) {
-    for (HighsInt j = 0; j < rank_deficiency; j++) {
-      ASM[i + j * rank_deficiency] = 0;
```

Comment:
Fair point, I'll remove this `malloc`. I wrote it (in debug code) before I knew that a pointer could be created as `vector.data()`